### PR TITLE
Package libyajl.so.1 as part of the Amazon AMI build

### PR DESCRIPTION
**Pull request by dhrupadb**

### CODE REVIEW

File: rpm/stackdriver-agent.spec [hunk 1]
```diff
@@ -110,6 +111,8 @@ BuildRequires: hiredis-devel
 %endif
 %if %{has_yajl}
 BuildRequires: yajl-devel
+%endif
+%if %{has_yajl} && ! %{bundle_yajl}
```
Comment (igorpeshansky) [inline rpm/stackdriver-agent.spec]: You can fold this into the previous `%if %{has_yajl}` and avoid the `&&`.
That said, you need to initialize `%{bundle_yajl}` via `%define bundle_yajl 0` around line 17.

File: rpm/stackdriver-agent.spec [hunk 2]
```diff
@@ -269,6 +272,13 @@ rm -rf %{buildroot}%{_prefix}/include/curl %{buildroot}%{_prefix}/lib/libcurl*
 %{__rm} -f %{buildroot}%{_prefix}/man/man5/%{programprefix}collectd-snmp.5*
 %{__rm} -f %{buildroot}%{_prefix}/bin/stackdriver-utils_vl_lookup_test
 
+%if %{bundle_yajl}
+mkdir -p %{buildroot}%{_libdir}/yajl/
+cp /usr/lib64/libyajl.so.1 %{buildroot}%{_libdir}/yajl/
+ln -s %{buildroot}%{_libdir}/yajl/libyajl.so.1 %{_libdir}/libyajl.so.1
```
Comment (igorpeshansky) [inline rpm/stackdriver-agent.spec]:
```sh
cd %{buildroot}%{_libdir} && ln -s yajl/libyajl.so.1 .
```
You want that link to be relative, not absolute.

File: rpm/stackdriver-agent.spec [hunk 3]
```diff
@@ -317,13 +327,23 @@ fi
 %{_libdir}/libmongoc-priv.*
 %{_libdir}/pkgconfig/*
 
+%if %{bundle_yajl}
+%{_libdir}/yajl/libyajl.so.1
```
Comment (igorpeshansky) [inline rpm/stackdriver-agent.spec]:
```specfile
%dir %{_libdir}/yajl
```

File: rpm/stackdriver-agent.spec (hunk 2 above)
Comment (igorpeshansky) [reply rpm/stackdriver-agent.spec]: Actually, even simpler:
```sh
ln -s -t %{buildroot}%{_libdir} yajl/libyajl.so.1
```
(no need for the subshell in that case).

File: rpm/stackdriver-agent.spec [hunk 4]
```diff
@@ -317,13 +328,24 @@ fi
 %{_libdir}/libmongoc-priv.*
 %{_libdir}/pkgconfig/*
 
+%if %{bundle_yajl}
+%dir %{_libdir}/yajl/
+%{_libdir}/libyajl.so.1
+%{_libdir}/yajl/libyajl.so.1
```
Comment (igorpeshansky) [inline rpm/stackdriver-agent.spec]: Let's move this above the symlink...

Comment (dhrupadb) [reply rpm/stackdriver-agent.spec]: Done.

File: rpm/stackdriver-agent.spec (hunk 1 above)
Comment (dhrupadb) [reply rpm/stackdriver-agent.spec]: It worked. Confirmed on Centos 6 and 7.

File: rpm/stackdriver-agent.spec (hunk 3 above)
Comment (dhrupadb) [reply rpm/stackdriver-agent.spec]: done.

File: rpm/stackdriver-agent.spec (hunk 2 above)
Comment (dhrupadb) [reply rpm/stackdriver-agent.spec]: Done.

File: rpm/stackdriver-agent.spec (hunk 1 above)
Comment (igorpeshansky) [reply rpm/stackdriver-agent.spec]: My main concern was actually Amazon Linux... :smile:

**Comment by igorpeshansky** [comment]
dhrupadb was at Google at the time.

**Pull request merged**